cmd/vcluster/cmd/start.go

```diff
@@ -299,6 +300,14 @@ func ExecuteStart(options *context2.VirtualClusterOptions) error {
 		}
 	}()
 
+	// check apiservice for metrics server
+	go func() {
+		err := metricsapiservice.RegisterOrDeregisterAPIService(ctx.Context, ctx.Options, virtualClusterConfig)
+		if err != nil {
+			klog.Fatalf("Error registering metrics apiservice: %v", err)
+		}
+	}()
```

Comment:
Alternatively, we can refactor slightly to ignore some errors to not care about concurrent execution. This would be a simpler code, but potentially more brittle wrt future changes, and also less efficient.

Reply:
Right, I think this call can be made inside the leader election blocks right below this section. Also instead of fatal we can just print out the warning/error messages and let things continue – because not registering the `apiservice` is only going to cause metrics not being proxied properly but not fatal to running vcluster itself. WDYT @matskiv ?

Reply:
Yes, I agree, let's move it to the code block executed by the leader, or the one executed when the leader election is disabled. I would keep it as fatal - there are retries to handle random network issues, but otherwise, it should not fail, so IMHO fatal allows us to uncover problems sooner (e.g. new version changes API).